airflow/config_templates/config.yml

```
@@ -450,6 +450,21 @@ core:
       type: string
       default: ~
       example: 'http://localhost:8080'
+    test_connection:
+      description: |
+        The ability to allow testing connections across Airflow UI, API and CLI.
+        Supported options: Disabled, Enabled, Hidden. Default: Disabled
+        Disabled - Disables the test connection functionality and disables the Test Connection button in UI.
+        Enabled - Enables the test connection functionality and shows the Test Connection button in UI.
+        Hidden - Disables the test connection functionality and hides the Test Connection button in UI.
+        Before setting this to Enabled, make sure that you review the users who are able to add/edit
+        connections and ensure they are trusted. Connection testing can be done maliciously leading to
+        undesired and insecure outcomes.
+      version_added: 2.6.3
```

```suggestion
      version_added: 2.7.0
```

I think (but I would love to hear what others think) - it's a bit too disruptive to release it in PATCHLEVEL release. I see this one as a long-term decision that should impact whether test connection is enabled or not, and there is no immediate need to add it in patchlevel version IMHO.

---

That makes sense, will keep it for 2.7.0 unless someone says we need it for 2.6.3.  Will make this change in my next commit.